src/main/java/com/google/devtools/build/lib/remote/circuitbreaker/CircuitBreakerFactory.java

```diff
@@ -16,13 +16,14 @@
 import com.google.common.collect.ImmutableSet;
 import com.google.devtools.build.lib.remote.Retrier;
 import com.google.devtools.build.lib.remote.common.CacheNotFoundException;
+import com.google.devtools.build.lib.remote.common.OutOfRangeException;
 import com.google.devtools.build.lib.remote.options.RemoteOptions;
 
 /** Factory for {@link Retrier.CircuitBreaker} */
 public class CircuitBreakerFactory {
 
   public static final ImmutableSet<Class<? extends Exception>> DEFAULT_IGNORED_ERRORS =
-      ImmutableSet.of(CacheNotFoundException.class);
+      ImmutableSet.of(CacheNotFoundException.class, OutOfRangeException.class);
```

Comment:
I think we are not moving at right direction for handling ignored errors here. Since we already have the [logic](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/remote/RemoteRetrier.java;l=47;drc=060fea547102b8bdb31b9b5a6396208227bf6c53) to determine whether to retry on certain errors, maybe we could reuse that here?

Reply:
updated